### ISSUE TITLE
Docker: Wrong error info returned when delete a committed images

### ISSUE THREAD

**Issue by hgangwx** [moby/moby#18256]
[Topo]
1 controller, 2 network nodes, 6 compute nodes, all in ubuntu 14.04
(openstack version is 2015.1.2, linux kernel version is 3.19.0-31, docker version 1.8.3)

```bash
root@ah:~# uname -a
Linux ah.container13 3.19.0-31-generic #36~14.04.1-Ubuntu SMP Thu Oct 8 10:21:08 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
root@ah:~#
root@ah:~# dpkg -l | grep neutron
ii neutron-common 1:2015.1.2-0ubuntu1~cloud0 all Neutron is a virtual network service for Openstack - common
ii neutron-plugin-ml2 1:2015.1.2-0ubuntu1~cloud0 all Neutron is a virtual network service for Openstack - ML2 plugin
ii neutron-server 1:2015.1.2-0ubuntu1~cloud0 all Neutron is a virtual network service for Openstack - server
ii python-neutron 1:2015.1.2-0ubuntu1~cloud0 all Neutron is a virtual network service for Openstack - Python library
ii python-neutron-fwaas 2015.1.2-0ubuntu2~cloud0 all Firewall-as-a-Service driver for OpenStack Neutron
ii python-neutronclient 1:2.3.11-0ubuntu1.2~cloud0 all client - Neutron is a virtual network service for Openstack
root@ah:~#
```

```bash
root@quasarsdnve1:~# docker version
Client:
 Version: 1.8.3
 API version: 1.20
 Go version: go1.4.2
 Git commit: f4bf5c7
 Built: Mon Oct 12 05:37:18 UTC 2015
 OS/Arch: linux/amd64

Server:
 Version: 1.8.3
 API version: 1.20
 Go version: go1.4.2
 Git commit: f4bf5c7
 Built: Mon Oct 12 05:37:18 UTC 2015
 OS/Arch: linux/amd64
root@quasarsdnve1:~#
```

[Description and expect result]
No wrong error info returned when delete a committed images

[Reproduceable or not]
reproduceable

[Recreate Steps]
1)pull a new `ewindisch/cirros` image:
```bash
root@ah:~# docker pull ewindisch/cirros
Using default tag: latest
latest: Pulling from ewindisch/cirros
d634beec75db: Pull complete
387d7eeeeedb: Pull complete
e3c5fde0316e: Pull complete
b1e4418e356f: Pull complete
494d8decc1f8: Pull complete
173efbcd07f3: Pull complete
Digest: sha256:b069d02b29811f81c2f87d7f994a158f5eb304d0a0e49f7217560e0db285cbe7
Status: Downloaded newer image for ewindisch/cirros:latest
root@ah:~#
root@ah:~# docker images
REPOSITORY TAG IMAGE ID CREATED VIRTUAL SIZE
wlwwu/ubuntu_iperf latest 41f404bfca01 5 days ago 252.5 MB
cirros latest 21bb29681c97 4 weeks ago 7.698 MB
ewindisch/cirros latest 173efbcd07f3 9 months ago 7.698 MB
root@ah:~#
```

2) run a container with the ewindisch/cirros images, and create a dir in the container:

```bash
root@ah:~# docker run -it ewindisch/cirros /bin/sh
/ # mkdir test_dir
/ # ls
bin dev home lib media old-root proc run sys tmp var
boot etc init linuxrc mnt opt root sbin test_dir usr
/ # exit
root@ah:~#
root@ah:~#
root@ah:~# docker ps -a | more
CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS
      NAMES
53295329ff55 ewindisch/cirros "/bin/sh" 22 seconds ago Exited (0) 8 seconds ago
```

3) commit modification of the container to image ewindisch/cirros:

```bash
root@ah:~# docker commit 53295329ff55 ewindisch/cirros
1be19b7c1296bdaccbc678e358a21dc987102f065cf4f3fe14b85ef7a106170b
root@ah:~#
root@ah:~#
```

4)also commit modification of the container to a new image ewindisch_cirros_1:

```bash
root@ah:~# docker commit 53295329ff55 ewindisch_cirros_1
875b087a1ebeef2060f6fcb35799f882577bfa5bc3dd87579f4aa909329f50e6
root@ah:~#
root@ah:~#
root@ah:~# docker images
REPOSITORY TAG IMAGE ID CREATED VIRTUAL SIZE
ewindisch_cirros_1 latest 875b087a1ebe 5 seconds ago 7.698 MB
ewindisch/cirros latest 1be19b7c1296 12 seconds ago 7.698 MB
wlwwu/ubuntu_iperf latest 41f404bfca01 5 days ago 252.5 MB
cirros latest 21bb29681c97 4 weeks ago 7.698 MB
root@ah:~# docker ps -a | more
CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS
            NAMES
53295329ff55 ewindisch/cirros "/bin/sh" About a minute ago Exited (0) About a minute ago
===> note, the image ewindisch/cirros is used by a container, but the new image ewindisch_cirros_1 is not used
```

5) try to remove the new image ewindisch_cirros_1, error info returned, says that the image is used, can not delete(actually it is not used):
```bash
root@ah:~# docker rmi ewindisch_cirros_1
Error response from daemon: Conflict, cannot delete 173efbcd07f3 because the container 53295329ff55 is using it, use -f to force
Error: failed to remove images: [ewindisch_cirros_1] >>>>>>>wrong error info, ISSUE
```

6)show the docker images again, found that the new image ewindisch_cirros_1 is actually removed:

```bash
root@ah:~# docker images
REPOSITORY TAG IMAGE ID CREATED VIRTUAL SIZE
ewindisch/cirros latest 1be19b7c1296 48 seconds ago 7.698 MB
wlwwu/ubuntu_iperf latest 41f404bfca01 5 days ago 252.5 MB
cirros latest 21bb29681c97 4 weeks ago 7.698 MB
root@ah:~#
```

[Configration]
reproduceable bug, no need

[logs]
reproduceable bug, no need

[Root cause anlyze or debug inf]
reproduceable bug

[Attachment]
None

**Comment by coolljt0725** [comment]
I think #17108 could fix this issues

**Comment by thaJeztah** [comment]
@coolljt0725 can you add a `fixes #....` to your PR?

**Comment by thaJeztah** [comment]
PR https://github.com/docker/docker/pull/17108 was just merged, @hgangwx you should be able to test if this issue is resolved by testing a build from master; https://master.dockerproject.org (don't test it on a production system though!)

**Comment by hgangwx** [comment]
@thaJeztah I verified on docker version 1.9.1, seems the ewindisch_cirros_1 image can be deleted correctly, but when I delete the original image "ewindisch/cirros", it returns an error info , and I observed that the tag and name of image "ewindisch/cirros" is deleted.
Can you check again? below is my operation log:
root@quasarsdnve1:/etc#  docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64
root@quasarsdnve1:/etc# 


root@quasarsdnve1:/etc# docker pull ewindisch/cirros
Using default tag: latest
latest: Pulling from ewindisch/cirros
d634beec75db: Pull complete 
387d7eeeeedb: Pull complete 
e3c5fde0316e: Pull complete 
b1e4418e356f: Pull complete 
494d8decc1f8: Pull complete 
173efbcd07f3: Pull complete 
Digest: sha256:b069d02b29811f81c2f87d7f994a158f5eb304d0a0e49f7217560e0db285cbe7
Status: Downloaded newer image for ewindisch/cirros:latest
root@quasarsdnve1:/etc# docker images
REPOSITORY                      TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
ewindisch/cirros                latest              173efbcd07f3        10 months ago       7.698 MB
root@quasarsdnve1:/etc# docker run -it ewindisch/cirros /bin/sh
/ # mkdir kkk
/ # exit
root@quasarsdnve1:/etc# docker ps -a | more
CONTAINER ID        IMAGE                   COMMAND                  CREATED             STATUS                 
       PORTS               NAMES
72e9ddc0a708        ewindisch/cirros        "/bin/sh"                18 seconds ago      Exited (0) 12 seconds a
go                         lonely_torvalds
root@quasarsdnve1:/etc#  docker commit 72e9ddc0a708 ewindisch/cirros
7bf0229c591f38c698e7219e5eff54f6ea6b76636c9d59f72f729d96f1a7e0d6
root@quasarsdnve1:/etc#  docker commit 72e9ddc0a708 ewindisch_cirros_1
55885900945dd9e00ecbefc5e8c2e566ee7500259ba576377675bb41732548e5
root@quasarsdnve1:/etc# docker images
REPOSITORY                      TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
ewindisch_cirros_1              latest              55885900945d        8 seconds ago       7.698 MB
ewindisch/cirros                latest              7bf0229c591f        32 seconds ago      7.698 MB
root@quasarsdnve1:/etc# docker rmi ewindisch_cirros_1   ---->succeeded.
Untagged: ewindisch_cirros_1:latest
Deleted: 55885900945dd9e00ecbefc5e8c2e566ee7500259ba576377675bb41732548e5
root@quasarsdnve1:/etc# docker images
REPOSITORY                      TAG                 IMAGE ID            CREATED              VIRTUAL SIZE
ewindisch/cirros                latest              7bf0229c591f        About a minute ago   7.698 MB
root@quasarsdnve1:/etc# docker rmi ewindisch/cirros    
Error response from daemon: conflict: unable to delete 173efbcd07f3 (must be forced) - image is being used by stopped container 72e9ddc0a708
Error: failed to remove images: [ewindisch/cirros]
root@quasarsdnve1:/etc# docker images
REPOSITORY                      TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
<none>                          <none>              173efbcd07f3        10 months ago       7.698 MB  >>>ISSUE
root@quasarsdnve1:/etc#

**Comment by coolljt0725** [comment]
@hgangwx  can you try the `master` branch? #17108 is not included in `1.9.1`
